### ISSUE TITLE
Arrow functions rule does not go well with Mocha tests

### ISSUE THREAD

**Issue by trygve-lie** [finn-no/eslint-config-finn#72]
These rules force the use of arrow functions. We also enforce running eslint as a pre hook to running tests and we lint our tests. We do also advice the use of Mocha as test framework.

Which leads us to Mocha tests with arrow functions. Issue is that Mocha discouraged the use of arrow functions in tests: https://mochajs.org/#arrow-functions 

I did actually encounter the problem they list in a test I just wrote. Others will also. We need to address this in a way.

Some ideas to do so:

a) - a more relaxed rule on the use of arrow functions
b) - two eslint rules, one for tests with a relaxed arrow function rule and one for non test code with a strong arrow functions rule.
c) - do not lint test code
d) - advice usage of another test framework

**Comment by SimenB** [comment]
I think Mocha only recommends against arrows because of `this` (which is an anti-pattern, IMO, but that's another discussion)

If you use `this` inside a test, eslint won't nag on you to make it into an arrow function, so I don't think this is an issue?

(We're also going to update our guidelines to say that Ava is OK to use for node projects)

**Comment by trygve-lie** [comment]
Yeah. With arrow functions one loose the context of the Mocha object in the test.

If you write an arrow function and want to access the Mocha context you will not be able too. It fails. So you need to use an old fashion function and the eslint rule will not complain if there is a `this` in there.

What one end up with is quickly a bunch of tests where one use arrow functions and then one might have one or two tests where one use a plain old function. Thats a small gotcha there.

I read the Mocha doc as; Mocha is not designed to work with arrow functions, they might work and you will loose context. No guarantee from us.

**Comment by SimenB** [comment]
The only time you "need" the mocha context is to set `timeout`, right? And if you need it, just use `function` for that single test.

IMO: if you want to share some object, put it in a `let` above `beforeEach` and assign to it, don't depend on mocha handling `this` for you. I'm not a fan of implicit APIs, and we should try to minimize it's usage.
